ext/gd/tests/func.inc

```diff
@@ -178,3 +155,23 @@ function trycatch_dump(...$tests) {
     }
 }
 
+/**
+ * Calculate the mean squared error (in range 0..255**2)
+ */
+function mse(GdImage $image1, GdImage $image2): float
+{
+    assert(imageistruecolor($image1) && imageistruecolor($image2));
+    assert(imagesx($image1) === imagesx($image2) && imagesy($image1) === imagesy($image2));
+    $e = $count = 0;
+    for ($j = 0, $m = imagesy($image1); $j < $m; $j++) {
+        for ($i = 0, $n = imagesx($image1); $i < $n; $i++) {
+            $c1 = imagecolorat($image1, $i, $j);
+            $c2 = imagecolorat($image2, $i, $j);
+            $e += ((($c1 >> 16) & 255) - (($c2 >> 16) & 255)) ** 2
+                + ((($c1 >>  8) & 255) - (($c2 >>  8) & 255)) ** 2
+                + ((($c1 >>  0) & 255) - (($c2 >>  0) & 255)) ** 2;
```

Comment:
Just stumbled upon https://github.com/libgd/libgd/pull/751, where the differences are divided by the maximum value, thus yielding values in range [0, 1] (instead of [0, 255]), so the squares would stay in range [0, 1] (instead of [0, 65025]), as would the final result of `mse()`. That might be even better (at least wrt possibly adding alpha support later).